### PR TITLE
[RTM] Bypass the maintenance mode if logged in to the back end

### PR DESCRIPTION
**Pull request by leofeyer**
This is the follow-up PR to #283.

### CODE REVIEW

**Comment by aschempp** [comment]
I intentionally did not name the parameter something with session, because it is not related to the (Symfony) session. In regards to the new image settings, maybe it should be `contao.user.bind_to_ip`?

The config would then make sense:

```yaml
contao:
    user:
        bind_to_ip: true
```

**Comment by aschempp** [comment]
there could potentially be more user configuration in the future

**Comment by leofeyer** [comment]
You cannot bind a user to an IP address. Only a session.

File: src/Resources/contao/library/Contao/Config.php [hunk 1]
```diff
@@ -513,6 +513,7 @@ protected static function loadParameters()
 		}
 
 		$GLOBALS['TL_CONFIG']['debugMode'] = $container->getParameter('kernel.debug');
+		$GLOBALS['TL_CONFIG']['disableIpCheck'] = $container->getParameter('contao.security.disable_ip_check');
```
Comment (aschempp) [inline src/Resources/contao/library/Contao/Config.php]: You can move this to the array above now that we don't need inversion. Dunno why `debugMode` is not up there…

**Comment by aschempp** [comment]
Hmm. Actually we do not need the path validation in the maintenance bundle! Our backend routes should have the respective attribute ?

File: src/Resources/contao/library/Contao/Config.php (hunk 1 above)
Comment (leofeyer) [inline src/Resources/contao/library/Contao/Config.php]: Fixed in da4ebe63f09d9edad4bc6d32729e372e398cf9c4.

**Comment by leofeyer** [comment]
Besides the fact that this would be more work, it would also not apply to custom back end routes added by third party developers.

**Comment by aschempp** [comment]
That's true, but it would allow for more fine-grained control. It would also mean it still perfectly works if I rename the backend route path.

**Comment by leofeyer** [comment]
If you rename the back end route, you can adjust the Lexik config as well ;)

**Comment by leofeyer** [comment]
Also, was it not you saying that using route attributes is magic that should be avoided? This is a case where we could avoid it.

**Comment by leofeyer** [comment]
Your suggestion is actually independent from this PR; if we should decide to use the route attribute, we'll have to adjust the standard edition. I have therefore merged this PR in cd2ebbe02f32a43dab2740abf11c100c2aa954c9.

**Pull request merged**